File: leetcode-js/011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    if (height.length === 0 || height.length === 1) {
        return 0;
    }

    let left = 0;
    let right = height.length - 1;
    let area = 0;

    while (left < right) {
        area = Math.max(area, Math.min(height[left], height[right]) * (right - left));
        if (height[left] > height[right]) {
            right -= 1;
        } else {
            left += 1;
        }
    }

    return area;
};

File: leetcode-java/011-container-with-most-water.java
public class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxArea = 0;

        while (left < right) {
            int h = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, h * (right - left));
            while (height[left] <= h && left < right) {
                left++;
            }
            while (height[right] <= h && left < right) {
                right--;
            }
        }

        return maxArea;
    }
}
